plugin-cgroupv1/tests/oar2.rs

```diff
@@ -0,0 +1,292 @@
+    };
+    use plugin_cgroupv1::{Oar2Config, Oar2Plugin};
+    use tempfile::tempdir;
+
+    /// This test ensure the plugin startup correctly, with the expected source based on OAR Mocks created during the test.
+    /// It also verifies the registered metrics and their units.
+    #[test]
+    fn test_startup() -> Result<(), Box<dyn std::error::Error>> {
+        let mut plugins = PluginSet::new();
+        let temp_dir = tempdir().unwrap();
+        let path = temp_dir.path();
+
+        let mocks = vec![
+            OarMock {
+                submitter: "somesubmitter",
+                job_id: "1234",
+                cgroup_files: vec![
+                    CgroupFile::CpuAcctUsage(123456789),
+                    CgroupFile::MemoryUsageInBytes(987654321),
+                ],
+            },
+            OarMock {
+                submitter: "somesubmitter",
+                job_id: "5678",
+                cgroup_files: vec![
+                    CgroupFile::CpuAcctUsage(123456789),
+                    CgroupFile::MemoryUsageInBytes(987654321),
+                ],
+            },
+            OarMock {
+                submitter: "anothersubmitter",
+                job_id: "666",
+                cgroup_files: vec![
+                    CgroupFile::CpuAcctUsage(123456789),
+                    CgroupFile::MemoryUsageInBytes(987654321),
+                ],
+            },
+        ];
+        create_oar_mocks(path, mocks)?;
+
+        let source_config = Oar2Config {
+            path: path.to_path_buf(),
+            poll_interval: Duration::from_secs(1),
+        };
+        plugins.add_plugin(PluginInfo {
+            metadata: PluginMetadata::from_static::<Oar2Plugin>(),
+            enabled: true,
+            config: Some(config_to_toml_table(&source_config)),
+        });
+
+        let startup_expectations = StartupExpectations::new()
+            .expect_metric::<u64>("cpu_time_delta", PrefixedUnit::nano(Unit::Second))
+            .expect_metric::<u64>("memory_usage", Unit::Byte)
+            .expect_source("oar2-plugin", "somesubmitter_1234")
+            .expect_source("oar2-plugin", "somesubmitter_5678")
+            .expect_source("oar2-plugin", "anothersubmitter_666");
+
+        let agent = agent::Builder::new(plugins)
+            .with_expectations(startup_expectations)
+            .build_and_start()
+            .unwrap();
+
+        agent.pipeline.control_handle().shutdown();
+        agent.wait_for_shutdown(Duration::from_secs(10)).unwrap();
+
+        Ok(())
+    }
+
+    /// This test performs lot of different tests in order to verify the plugin run correctly at runtime:
+    /// It's orchestrated in a way to test different scenario: first poll and other polls, since first poll will not have measurements for CounterDiff metrics.
+    /// It also ensure cgroup that pop up during Alumet's runtime are correctly collected too.
+    #[test]
+    fn test_runtime() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Since you have used `unwrap` a lot, you can replace `?` by unwrap or expect and it'll be fine.
Another way would be to use `anyhow` Errors, which would be a bit more easy to write/read and would provide nice backtraces.